### DISCUSSION TITLE
Sort library by [%<artist>%]

### DISCUSSION THREAD

**Discussion by takaji10** [TT-ReBORN/Georgia-ReBORN#12] [Q&A] [answered]
I'd like to sort the library tree by `[%<artist>%]`, which would place each artist on its own line. Currently, `%artist%` fields with multiple artists (separated by ;) appears as its own line, which makes the list longer and messier.

See this example:
![image](https://user-images.githubusercontent.com/1534470/124939439-647a5b80-e009-11eb-9518-e52144de7307.png)

Hopefully it's possible?

Thanks for the great work on this layout! :)

**Comment by TT-ReBORN** [comment]
Hi takaji10,

the library actually does sort artist by default. It seems you have not optimal tagged your albums.

Of course the library will display correctly as shown in your example, because if all other co-artists are 
in the %artist% tag, it can not differentiate the other co-artist. It is handled like a full artist name...

Here is how to tag it correctly:

%artist% = Emmanuel Holterbach 
and 
%album artist% = Emmanuel Holterbach, Etienne Coussirat, Yannick Dauby

This will show artist & co-artists in the playlist and the artist as one line in the library.

When you have tagged it correctly, you need to reinitialize the Media Library.
1. Preferences ->Media Library , remove all your music folders, apply and reload foobar. 
2. Now add again all music folders in your Media Library, apply and reload foobar. 

-TT

**Reply by takaji10** [reply]
Thanks for your feedback!

I don't think that's the optimal solution, because in this case there is an album with 17 tracks by "Emmanuel Holterbach" and three of those tracks featuring another artist "Michael Northam". Therefore it makes more sense for %album artist% to be "Emmanuel Holterbach" and not "Emmanuel Holterbach; Michael Northam" as I wouldn't want the album to show up with both artists as the album artist.

foobar2000 can actually handle multiple tags in a single tag. If you add `<>` in the tag, like `%<artist>%`, then foobar2000 will return artist values split by the semicolon that is used to append multiple different tags in a single tag.

**Reply by TT-ReBORN** [reply]
takaji10 , 

you're right about the multi tags, totally forgot about it, was quite late... =)
I have tested it and the right way to achieve this is quite the opposite what I've written.

---------------------------------------------------------------------------------

If it's the whole album:


%artist% = Emmanuel Holterbach, Etienne Coussirat, Yannick Dauby
%album artist% = Emmanuel Holterbach


On individual songs you tag the songs with the multiple artists e.g:

Song 1, 2, 3:

%artist% = Emmanuel Holterbach
%album artist% = Emmanuel Holterbach

Song 4 = co-artist:

%artist% = Emmanuel Holterbach; Etienne Coussirat
%album artist% = Emmanuel Holterbach

Song 5, 6, 7, 8, 9 = multiple artists:

%artist% = Emmanuel Holterbach; Etienne Coussirat; Yannick Dauby
%album artist% = Emmanuel Holterbach

Song 10, 11, 12 = other co artist:

%artist% = Yannick Dauby;
%album artist% = Emmanuel Holterbach

---------------------------------------------------------------------------------

Try it out for yourself, right click on the song -> Properties ( Alt + Enter ).
Now it should show correctly in the playlist and in the library ( one line ).

Also in the library you need to right click -> View -> View by Album Artist

-TT

**Reply by takaji10** [reply]
Hi TT, yep, this is how I have my tagging set up right now, and I'm also viewing by album artist.

But it would be really nice to see only a single entry in the library tree for each artist. Using this method, it's not possible to, say, view all albums containing the artist "Alun Francis" because the artist is being displayed as a new line for each album they're a part of:

![image](https://user-images.githubusercontent.com/1534470/125155767-4d02b600-e162-11eb-9cec-12e9182e16d1.png)

If using `[%<artist>%]` is not a sort method that you want to support, could you kindly let me know where I can change this or add another view option in the code?
Or if you do want to support this view, I would revise your code for "View by artist" and use this, otherwise the current view is technically more like "View by artist, grouped by album." You could otherwise make a new entry "View by artist (separated)" or something like this.
(I can also take a bit of time to see if I can find it myself :) )

Anyways, thank you!

**Reply by TT-ReBORN** [reply]
Hi takaji10,

Look at these, it works. In the library -> View by Album Artist:

![Library](https://i.ibb.co/Xt6p1Sr/Library.jpg)

![Tagging](https://i.ibb.co/Gn0yyBN/Tagging.jpg)

-TT

**Reply by TT-ReBORN** [reply]
Updated my previous post, look at the example.

**Reply by takaji10** [reply]
I'm not sure I understand your tagging rationale. If all tracks are by Alun Francis and the Saarbrucken Radio SO, why is there a need for an album artist? And why only the conductor?

foobar2000 encourages the use of a semicolon to separate multiple values in a tag (see https://wiki.hydrogenaud.io/index.php?title=Foobar2000:Properties, "Multivalue fields"), so this should also be used instead of the comma separator.

Your example only works if %album artist% is set up like in your example - if it was "Alun Francis; Saarbrucken Radio SO" then it would display as "Alun Francis, Saarbrucken Radio SO" and not as separate artists.

I tried to add an extra line after line 930 in \georgia\js\CaTRoX_QWR\Panel_Library.js like this:
`window.GetProperty(prefix + "View 08: Name // Pattern", "View by Artist (Separated) // %<artist>%|%album%|[[%discnumber%.]%tracknumber%. ][%track artist% - ]%title%")`
(and remembering to add a comma at the end of line 930)
But it didn't work for me when I restarted foobar2000. How could I create an entry that would sort by `%<artist>%`?

Thanks :)

**Reply by TT-ReBORN** [reply]
Ok, 

just use the Musicbrainz tagging formula, this also works.

Here is the foo_musicbrainz component:
https://www.foobar2000.org/components/view/foo_musicbrainz

I tried it out on your 4 albums and it works.
Select the whole album, right click -> Tagging -> Get tags from Musicbrainz

Sometimes if the album isn't automatically found, you need to manually add the ID of the album, e.g:
https://musicbrainz.org/artist/16592ac0-7fe5-4e24-9ac0-2deb74a66417

Click on the album, then details, there you see the MBID, copy and paste and update tags.
This will automatically fill and update all your tags, now click on Library and there you have it 
hopefully as you would like, compact in one line...

-TT

**Reply by takaji10** [reply]
I just tried it, and the results are not ideal. Compare this screenshot with my tagging on the left and Musicbrainz on the right:

![image](https://user-images.githubusercontent.com/1534470/125327109-0d46f480-e343-11eb-8b4d-6ab4c29f7698.png)

Musicbrainz has made the composer as the artist, and made the album artist as the composer and the performers. Many tagging discussions/private music trackers specify that the composer is never the track artist, and this solution drops the semicolon convention to separate artists.

This solution also doesn't make the library tree view as I'm trying to make it look. If it did, I would expect to see:
```
Alun Francis
Rundfunk‐Sinfonieorchester Saarbrücken
```
but instead am seeing:
![image](https://user-images.githubusercontent.com/1534470/125329580-f2c24a80-e345-11eb-8df2-ad78969ee1bf.png)

This is how using `[%<artist>%]` looks in foo_facets (bottom left) and foo_simplaylist - each artist has its own line, and clicking on Alun Francis' name reveals all albums that contain him as an artist, so it's easy to see which albums he belongs to in a single playlist:
![image](https://user-images.githubusercontent.com/1534470/125329247-90694a00-e345-11eb-82a4-148d59cbfeb3.png)

I don't think that the answer here is to re-tag my library according to Musicbrainz convention - Georgia-ReBORN should support the tagging system I'm using, especially since I consider it to be pretty close to accepted tagging standards and I'm certain many people are using this standard of tagging.

Couldn't you change the "View by Artist" grouping option to use `[%<artist>%]` instead of `[%artist%]`, and also add a "View by Composer" that uses `[%<composer>%]`? The composer tag shows as a default tagging field in foobar2000 properties so it makes sense that there would also be an option to sort by this. Using `[%<artist>%]` over `[%artist%]` would satisfy those users that want a single line for each artist (who are using semicolon to separate multiple values in a tagging field) and not change anything for those users who are using tagging standards like Musicbrainz or just using commas between multiple values.

**Comment by kbuffington** [comment, marked as answer]
Library panel already supports this, it's just not set by default for the Artist view. Just update the artist entry to use `%<artist>%`:

![image](https://user-images.githubusercontent.com/2282004/125338104-6730c580-e315-11eb-968e-bfaf432d73ff.png)

No retagging necessary.

**Reply by TT-ReBORN** [reply]
OMG, I totally forgot about to update the SMP configuration properties, sometimes it's that simple ;-)

Thanks mate!

**Reply by takaji10** [reply]
Perfect! This is what I was looking for. Thanks @kbuffington and @TT-ReBORN!

@TT-ReBORN , could I request an additional "View by Composer" option to make it into the next code update? :)

**Reply by TT-ReBORN** [reply]
These two?

`"View by Composer // %composer%|%album%|[[%discnumber%.]%tracknumber%. ][%track artist% - ]%title%"),`

`"View by Composer - Album // %<composer>%|%album%|[[%discnumber%.]%tracknumber%. ][%track artist% - ]%title%"),`

**Reply by takaji10** [reply]
I think only the first option is needed, but up to you!
